src/components/accession2/WithdrawModal.tsx

```diff
@@ -0,0 +1,273 @@
+      setWithdrawAllSelected(false);
+    }
+    if (record) {
+      setRecord({
+        ...record,
+        withdrawnQuantity: { quantity: record.withdrawnQuantity?.quantity || 0, units: newValue.value },
+      });
+    }
+  };
+
+  const onSelectAll = (id: string, value: boolean) => {
+    if (!withdrawAllSelected) {
+      setRecord({
+        ...record,
+        withdrawnQuantity: {
+          quantity: accession.remainingQuantity?.quantity || 0,
+          units: accession.remainingQuantity?.units || 'Grams',
+        },
+      });
+    } else {
+      setRecord({
+        ...record,
+        withdrawnQuantity: {
+          quantity: 0,
+          units: 'Grams',
+        },
+      });
+    }
+
+    setWithdrawAllSelected(!withdrawAllSelected);
+  };
+
+  const onChangeDate = (id: string, value?: any) => {
+    const date = new Date(value).getTime();
+    const now = Date.now();
+    if (isNaN(date) || date > now) {
+      return;
+    } else {
+      onChange(id, value);
+    }
+  };
+
+  const onCloseHandler = () => {
+    setWithdrawAllSelected(false);
+    setRecord(newWithdrawal);
+    onClose();
+  };
+
+  return (
+    <DialogBox
+      onClose={onCloseHandler}
+      open={open}
+      title={strings.WITHDRAW_SEEDS}
+      size='medium'
+      middleButtons={[
+        <Button label={strings.CANCEL} type='passive' onClick={onCloseHandler} priority='secondary' key='button-1' />,
+        <Button onClick={saveWithdrawal} label={strings.WITHDRAW} key='button-2' />,
+      ]}
+    >
+      <Grid item xs={12} textAlign='left'>
+        <Grid item xs={12}>
+          <Select
+            label={strings.PURPOSE}
+            placeholder={strings.SELECT}
+            options={WITHDRAWAL_PURPOSES}
+            onChange={(value: string) => onChange('purpose', value)}
+            selectedValue={record?.purpose}
+            fullWidth={true}
+            readonly={true}
+          />
+        </Grid>
+        {record.purpose === 'Viability Testing' ? (
+          <>
+            <Select
+              label={strings.TEST_TYPE}
+              placeholder={strings.SELECT}
+              options={['Lab', 'Nursery']}
+              onChange={(value: string) => onChangeViabilityTesting('testType', value)}
+              selectedValue={viabilityTesting?.testType}
+              fullWidth={true}
+              readonly={true}
+            />
+            <Select
+              label={strings.SUBSTRATE}
+              placeholder={strings.SELECT}
+              options={['Nursery Media', 'Agar Petri Dish', 'Paper Petri Dish', 'Other']}
+              onChange={(value: string) => onChangeViabilityTesting('substrate', value)}
+              selectedValue={viabilityTesting.substrate}
+              fullWidth={true}
+              readonly={true}
+            />
+            <Select
+              label={strings.TREATMENT}
+              placeholder={strings.SELECT}
+              options={['Soak', 'Scarify', 'GA3', 'Stratification', 'Other']}
```

Comment:
best to export this as a type from src/types? same for other hardcoded lists in this file like test types